src/adapter/graphql/src/queries/datasets/dataset_endpoints.rs

```diff
@@ -0,0 +1,132 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use opendatafabric as odf;
+use url::Url;
+
+use crate::prelude::*;
+use crate::queries::*;
+
+pub struct DatasetEndpoints<'a> {
+    owner: &'a Account,
+    dataset_handle: odf::DatasetHandle,
+}
+
+#[Object]
+impl<'a> DatasetEndpoints<'a> {
```

Comment:
Example of usage: https://github.com/kamu-data/kamu-cli/pull/570#issuecomment-2023307609